specification/loadtestservice/LoadTestService.Management/models.tsp

```diff
@@ -0,0 +1,258 @@
+using Azure.Core;
+using Azure.ResourceManager;
+using OpenAPI;
+using Autorest;
+using Azure.ResourceManager.Foundations;
+using Azure.ResourceManager.Private;
+
+namespace Microsoft.LoadTestService;
+
+@doc("LoadTest details.")
+model LoadTestResource is TrackedResource<LoadTestProperties> {
+  @doc("Load Test name")
+  @key("loadTestName")
+  @extension("x-ms-parameter-location", "method")
+  @path
+  @segment("loadTests")
+  name: string;
+
+  ...ManagedServiceIdentity;
+}
+
+@doc("Resources provisioning states.")
+enum ResourceState {
+  @doc("Deleted state.")
+  Deleted: "Deleted",
+
+  ...ResourceProvisioningState,
+}
+
+// can be renamed but since the type is already used in published docs, it will be a breaking change.
+@doc("Managed identity type to use for accessing encryption key Url.")
+enum type {
+  @doc("System assigned identity.")
+  SystemAssigned: "SystemAssigned",
+
+  @doc("User assigned identity.")
+  UserAssigned: "UserAssigned",
+}
+
+@doc("Details about the connection between the Batch service and the endpoint.")
+model EndpointDetail {
+  @visibility("read")
+  @doc("The port an endpoint is connected to.")
+  port?: int32;
+}
+
+@doc("A domain name and connection details used to access a dependency.")
+model EndpointDependency {
+  @doc("The domain name of the dependency. Domain names may be fully qualified or may contain a * wildcard.")
+  @visibility("read")
+  domainName?: string;
+
+  @doc("Human-readable supplemental information about the dependency and when it is applicable.")
+  @visibility("read")
+  description?: string;
+
+  @doc("The list of connection details for this endpoint.")
+  @visibility("read")
+  @extension("x-ms-identifiers", ["port"])
+  endpointDetails?: EndpointDetail[];
+}
+
+@doc("A collection of related endpoints from the same service for which the Batch service requires outbound access.")
+model OutboundEnvironmentEndpoint {
+  @doc("The type of service that Azure Load Testing connects to.")
+  @visibility("read")
+  category?: string;
+
+  @doc("The endpoints for this service to which the Batch service makes outbound calls.")
+  @visibility("read")
+  @extension("x-ms-identifiers", ["domainName"])
+  endpoints?: EndpointDependency[];
+}
+
+@doc("All identity configuration for Customer-managed key settings defining which identity should be used to auth to Key Vault.")
+model EncryptionPropertiesIdentity {
+  @doc("Managed identity type to use for accessing encryption key Url.")
+  type?: type;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "We need resourceId to be nullable."
+  @doc("User assigned identity to use for accessing key encryption key Url. Ex: /subscriptions/fa5fc227-a624-475e-b696-cdd604c735bc/resourceGroups/<resource group>/providers/Microsoft.ManagedIdentity/userAssignedIdentities/myId.")
+  resourceId?: string | null;
+}
+
+@doc("Key and identity details for Customer Managed Key encryption of load test resource.")
+model EncryptionProperties {
```

Comment:
Same as above. See built-in Customer Managed Key Encryption link in previous comment.

Reply:
The built in Encryption and Customer Managed key have additional field, here encryption only has identity and keyUrl whereas the inbuilt Encryption has configurationhas infrastructureEncryption, customerManagedKeyEncryption and  keyEncryptionKeyUrl properties.